tox.ini

```diff
@@ -64,17 +69,128 @@ allowlist_externals = env
 commands =
     env PATH="{env_bin_dir}" coverage run -m pytest {posargs:tests -ra --durations=20}
 
-[testenv:lint]
-base_python = py313,py312,py311,py310,py39
-use_develop = true
-deps =
-    pylint==3.2.7
-    pytest>=6
-    pytest-mock
-    importlib_metadata; python_version<"3.10"
-    iniconfig
+[testenv:pre-commit]
+description =
+  Run the quality checks under {basepython}; run as
+  `SKIP=check-id1,check-id2 tox r -e pre-commit` to instruct the underlying
+  `pre-commit` invocation avoid running said checks; Use
+  `tox r -e pre-commit -- check-id1 --all-files` to select checks matching IDs
+  aliases{:} `tox r -e pre-commit -- mypy --all-files` will run 3 MyPy
+  invocations, but `tox r -e pre-commit -- mypy-py313 --all-files` runs one.
 commands =
-    pylint {posargs:lektor tests}
+  {envpython} \
+    {[python-cli-options]byte-errors} \
+    {[python-cli-options]max-isolation} \
+    {[python-cli-options]warnings-to-errors} \
+    -m pre_commit \
+      run \
+      --color=always \
+      --show-diff-on-failure \
+      {posargs:--all-files}
+
+  # Print out the advice on how to install pre-commit from this env into Git:
+  -{envpython} \
+    {[python-cli-options]byte-errors} \
+    {[python-cli-options]max-isolation} \
+    {[python-cli-options]warnings-to-errors} \
+    -c \
+      'cmd = "{envpython} -m pre_commit install"; \
+      scr_width = len(cmd) + 10; \
+      sep = "=" * scr_width; \
+      cmd_str = "    $ \{cmd\}";' \
+      'print(f"\n\{sep\}\nTo install pre-commit hooks into the Git repo, run:\
+      \n\n\{cmd_str\}\n\n\{sep\}\n")'
+commands_post =
```

Comment:
What does all this in `commands_post` do?
(As best I can tell, it doesn't do anything for us?  Maybe its copypasta, or preparation for something more?)